seacatauth/client/provider/abc.py

```
@@ -0,0 +1,145 @@
+	async def initialize(self, app):
+		"""
+		Initialize provider. This is called by ClientService after all providers are registered.
+		"""
+		pass
+
+
+	@abc.abstractmethod
+	async def iterate_clients(
+		self,
+		page: int = 0,
+		limit: int = None,
```

Ruff (0.15.0)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)